src/main/scala/com/box/castle/core/leader/TaskManager.scala

```diff
@@ -59,20 +66,25 @@ class TaskManager(val leaderConfig: LeaderConfig, val committerConfigs: Iterable
       tasks
   }
 
-  private def matchTopicsRegex(topicsRegex: Regex, kafkaTopic: KafkaTopic, committerId: String, committerIds: Set[String]) = {
-    kafkaTopic.name match {
-      case topicsRegex(_*) => committerIds + committerId
-      case _ => committerIds
-    }
+  protected[leader] def matchTopicsRegex(topicsRegex: Regex, kafkaTopic: KafkaTopic, committerId: String, committerIds: Set[String]) = {
+      kafkaTopic.name match {
+        case topicsRegex(_*) => if(matchCommitterTopicFilter(kafkaTopic,committerId)) committerIds + committerId else committerIds
+        case _ => committerIds
+      }
   }
 
-  private def matchTopicsSet(topicsSet: Set[String], kafkaTopic: KafkaTopic, committerId: String, committerIds: Set[String]) = {
-    if (topicsSet.contains(kafkaTopic.name))
+  protected[leader] def matchTopicsSet(topicsSet: Set[String], kafkaTopic: KafkaTopic, committerId: String, committerIds: Set[String]) = {
+    if (topicsSet.contains(kafkaTopic.name) && matchCommitterTopicFilter(kafkaTopic, committerId))
       committerIds + committerId
     else
       committerIds
   }
 
+  protected[leader] def matchCommitterTopicFilter(kafkaTopic: KafkaTopic, committerId: String): Boolean = {
```

Comment:
It might be good to make function behavior consistent -- matchTopicsSet(...) returns a Set[String], while matchCommitterTopicFilter returns a Boolean.